app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -0,0 +1,85 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.annotation.StringRes
+import org.oppia.app.R
+import org.oppia.app.model.RatioExpression
+
+/** This class contains method that helps to parse string to ratio. */
+class StringToRatioParser {
+  private val invalidCharsRegex =
+    """^[\d\s/:]+$""".toRegex()
+  private val invalidRatioRegex =
+    """^(\s)*(\d+((\s)*:(\s)*\d+)+)(\s)*$""".toRegex()
+
+  /**
+   * Returns a [RatioParsingError] for the specified text input if it's an invalid ratio, or
+   * [RatioParsingError.VALID] if no issues are found. Note that a valid ratio returned by this method is guaranteed
+   * to be parsed correctly by [parseRatioFromString].
+   *
+   * This method should only be used when a user tries submitting an answer. Real-time error detection should be done
+   * using [getRealTimeAnswerError], instead.
+   */
+  fun getSubmitTimeError(text: String, numberOfTerms: Int): RatioParsingError {
+    if (!text.matches(invalidRatioRegex))
+      return RatioParsingError.INVALID_FORMAT
+    val ratio = parseRatioOrThrow(text)
+    return if (ratio.ratioComponentCount < numberOfTerms) {
```

Comment:
Is this correct? Shouldn't it be ``!=``? Please add a test to verify that too many terms also fails.

And, shouldn't this only be applied of numberOfTerms != 0? Please add a test for that, too.

Reply:
done